pkg/apis/validate/metadata.go

```diff
@@ -41,5 +43,15 @@ func ObjectMetadata(meta metav1.Object) *apis.FieldError {
 			Paths:   []string{"name"},
 		}
 	}
+
+	// metadata must not contain any label with tekton.dev/*
+	for k := range meta.GetLabels() {
+		if strings.HasPrefix(k, pipeline.GroupName) {
```

Comment:
Do we care/have an opinion on things like "*.tekton.dev" as well?

Reply:
This is meant to be Tekton internal, created and utilized by the pipeline itself, better to enforce rather than relying on `Task/Pipeline/TaskRun/PipelineRun` authors to follow recommendations, thoughts?

Reply:
@dlorenc good point. I would think we should reserve `tekton.dev` *and* most likely `pipelines.tekton.dev` (future-proof :sweat_smile: ), but given some [proposals](https://github.com/tektoncd/community/pull/147), we shouldn't restrict all the `*.tekton.dev` :upside_down_face:

Reply:
+1 on saving the extra namespace. In chains right now we're using "chains.tekton.dev", but only in an annotation.

Reply:
This PR is  restricting labels, annotations have all the freedom 😉